redis/src/main/scala/zio/redis/api/Scripting.scala

```diff
@@ -0,0 +1,119 @@
+package zio.redis.api
+
+import zio.{Chunk, ZIO}
+import zio.redis._
+import zio.redis.Input._
+import zio.redis.Output._
+import zio.redis.ResultBuilder.ResultOutputBuilder
+
+trait Scripting {
+  import Scripting._
+
+  /**
+   * Evaluates a Lua script
+   *
+   * Example of custom data:
+   * {{{
+   * final case class Person(name: String, age: Long)
+   *
+   * val encoder = Input.Tuple2(StringInput, LongInput).contramap[Person] { civ =>
+   *   (civ.name, civ.age)
+   * }
+   * val decoder = RespValueOutput.map {
+   *   case RespValue.Array(elements) =>
+   *     val name = elements(0) match {
+   *       case s @ RespValue.BulkString(_) => s.asString
+   *       case other                       => throw ProtocolError(s"$other isn't a string type")
+   *     }
+   *     val age = elements(1) match {
+   *       case RespValue.Integer(value) => value
+   *       case other                    => throw ProtocolError(s"$other isn't a integer type")
+   *     }
+   *     Person(name, age)
+   *   case other => throw ProtocolError(s"$other isn't an array type")
+   * }
+   * }}}
+   *
+   * @param script
+   *   Lua script
+   * @param keys
+   *   keys available through KEYS param in the script
+   * @param args
+   *   values available through ARGV param in the script
+   * @return
+   *   redis protocol value that is converted from the Lua type. You have to write decoder that would convert redis
+   *   protocol value to a suitable type for your app
+   */
+  def eval[K: Input, A: Input](
+    script: String,
+    keys: Chunk[K],
+    args: Chunk[A]
+  ): ResultOutputBuilder = new ResultOutputBuilder {
+    def returning[R: Output]: ZIO[RedisExecutor, RedisError, R] = {
+      val command = RedisCommand(Eval, EvalInput(implicitly[Input[K]], implicitly[Input[A]]), implicitly[Output[R]])
```

Comment:
If possible, add summoners for Input and Output. It'll make the call site cleaner.